plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -70,8 +71,11 @@ private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColum
                         List<Object> singleValues = new ArrayList<>();
                         List<String> rangeConjuncts = new ArrayList<>();
                         String predicate = null;
-
-                        for (Range range : ranges.getOrderedRanges()) {
+                        List<Range> orderedRanges = ranges.getOrderedRanges();
+                        if (rangesAreEquivalentToSingleValueNegation(orderedRanges)) {
+                            return null;
+                        }
+                        for (Range range : orderedRanges) {
```

Comment:
the `predicate = Joiner.on(" AND ").join(rangeConjuncts);` below is incorrect, isn't it?
Should be `" OR "` i guess?

is it what the bug report is actually about?

Reply:
Yes, the pointed code is the cause. However, Cassandra doesn't support `OR`.

Reply:
so we should remove that code -- first.
and limit pushdowns to single-value (or INs).

and this will implicitly solve the negation problem as well

Reply:
Ok so this would fix the issue, but we would lose pushdown of a single multivalued ranges, which is not that great.
I think that we need to detect when we should skip pushing down

Reply:
> Ok so this would fix the issue, but we would lose pushdown of a single multivalued ranges, which is not that great.

i didn't say that

depends how the code is written